setuptools/compat/py311.py

```diff
@@ -1,12 +1,24 @@
-import sys
+from __future__ import annotations
+
 import shutil
+import sys
+from typing import Any, Callable, TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from _typeshed import StrOrBytesPath, ExcInfo
```

Comment:
> Is there anyway we can simplify this?

The runtime fix is just the added conditional. Everything else is type hints.

I just combined imports, otherwise I could omit annotating the internal function and move `StrOrBytesPath` to a runtime alias with `StrPath` in `setuptools._path`.